src/Driver/PantherDriver.php

```diff
@@ -499,25 +557,31 @@ public function executeScript($script)
         $this->client->executeScript($script);
     }
 
-    public function executeAsyncScript($script)
-    {
-        $this->client->executeAsyncScript($script);
-    }
-
     /**
      * {@inheritdoc}
      */
     public function evaluateScript($script)
     {
-        throw new UnsupportedDriverActionException(sprintf('The %s::%s() method cannot be used.', self::class, __METHOD__), $this);
+        return $this->client->executeScript($script);
+    }
+
+    public function executeAsyncScript(string $script, array $arguments = []): void
+    {
+        $this->client->executeAsyncScript($script, $arguments);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function wait($timeout, $condition)
+    public function wait($timeoutInMilliseconds, $condition)
     {
-        return (bool) $this->client->wait($timeout)->until($condition);
+        if (!\is_callable($condition) || !$condition instanceof WebDriverExpectedCondition) {
```

Comment:
this does not fix the inconsistency with the interface, but i don't know if this can be done, since there is not string representation of a `WebDriverExpectedCondition`

Reply:
Actually, we can obtain a `callable` thanks to `WebDriverExpectedCondition::getApply()`, as the type is not checked at runtime, the main issue is related to static analysis tool if I understand correctly?

Reply:
Hi 👋 

I've just pushed a new implementation of this method, if you have time to check it 🙂